### DISCUSSION TITLE
Compilation error: `Error: Symbol’s function definition is void: -on`

### DISCUSSION THREAD

**Discussion by FJMMKM** [alphapapa/org-ql#427] [Q&A] [answered]
### OS/platform

Windows 11

### Emacs version and provenance

Emacs 27

### Emacs command

saving .emacs (use-package ...) called.

### Org version and provenance

Org 9

### org-ql package version and provenance

8.6

### Actions taken

compiling source

### Observed results

Compiling file c:/USR/FJM/APP/.emacs.d/elpa/org-ql-0.8.6/org-ql-completing-read.el at Wed May 22 19:21:49 2024
Entering directory ‘c:/USR/FJM/APP/.emacs.d/elpa/org-ql-0.8.6/’
org-ql-completing-read.el:27:1:Warning: reference to free variable ‘inherit’
org-ql-completing-read.el:27:1:Error: Symbol’s function definition is void: -on

Compiling file c:/USR/FJM/APP/.emacs.d/elpa/org-ql-0.8.6/org-ql-find.el at Wed May 22 19:21:55 2024
org-ql-find.el:32:1:Warning: reference to free variable ‘inherit’
org-ql-find.el:32:1:Error: Symbol’s function definition is void: -on

Compiling file c:/USR/FJM/APP/.emacs.d/elpa/org-ql-0.8.6/org-ql-search.el at Wed May 22 19:21:56 2024
org-ql-search.el:40:1:Warning: reference to free variable ‘inherit’
org-ql-search.el:40:1:Error: Symbol’s function definition is void: -on

Compiling file c:/USR/FJM/APP/.emacs.d/elpa/org-ql-0.8.6/org-ql-view.el at Wed May 22 19:21:57 2024
org-ql-view.el:43:1:Warning: reference to free variable ‘inherit’
org-ql-view.el:43:1:Error: Symbol’s function definition is void: -on

### Expected results

successful compilation

### Backtrace

```elisp
N/A
```

### Etc.

I looked at the code.  I am a lawyer.  I kind of get it.  If it were me it would a find & replace editing problem.

Seems like a nice package.  I would like to get it up and running.

**Comment by alphapapa** [comment, marked as answer]
Hello,

`Error: Symbol’s function definition is void: -on`: This error indicates that a dependency package is not installed.  If you install Org QL with `M-x package-install`, it will install things correctly.

Also, FYI, you should probably upgrade to Emacs 29.3; version 27 is rather old now, and some important issues are fixed in later versions.
